persistent_binderhub/files/jupyterhub/persistent_bhub_config.py

```diff
@@ -195,13 +195,13 @@ def start(self):
         #  because start command is run as an ENTRYPOINT and initcontainer's command overwrites it
         #  But start command will be executed in notebook container (because we dont define a custom command for it),
         #  so change will take place there, and on user's side, there is no problem
-        self.init_containers = [{
+        self.init_containers.append({
```

Comment:
Sorry, just not sure, but will `start()` be called only once per instance?

Reply:
I believe so, a subclass with `start`
```
def start(self):
     self.init_containers.append(another_container_spec)
     super().start()
```
should work IMO.

Reply:
I just am not sure, if it is possible that the same instance calls `start()` more than once. In this case, we would end up with 2 identical init containers. But if this is not possible, everything is fine

Reply:
IMO this should be okay (we can always revert this if something breaks :) )

This is how the subclassing would look like
```
In [1]: class A:  # KubeSpawner
   ...:     def __init__(self):
   ...:         self.a = list()
   ...:     def start(self):
   ...:         self.a.append('something')
   ...:

In [2]: class B(A):  # PersistentBinderSpawner
   ...:     def start(self):
   ...:         self.a.append('foo')
   ...:         super().start()
   ...:

In [3]: class C(B):
   ...:     def start(self):
   ...:         self.a.append('bar')
   ...:         super().start()
   ...:

In [4]: test = C()

In [5]: test.a
Out[5]: []

In [6]: test.start()

In [7]: test.a
Out[7]: ['bar', 'foo', 'something']
```

Reply:
Still to make my concerns clear. What I fear is the following:
```
test = C()
test.start()
...
test.start()
```

Reply:
ah yes, that would create copies.
But it's getting called once, assuming the custom spawner is set using
```
c.JupyterHub.spawner_class = CustomSpawner
```